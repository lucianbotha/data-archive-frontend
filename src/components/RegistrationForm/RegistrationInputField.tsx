import * as React from "react";

interface IDAInputProps {
  label?: string;
  name?: string;
  onChange?: any;
  placeholder?: string;
  type?: string;
  value?: string;
}

const RegistrationInputField = (props: IDAInputProps) => {
  const { label, name, onChange, placeholder, type, value } = props;
  return (
    <div className={"field"}>
      <label htmlFor={`${name}+${label}`} className={"label"}>
        {label}:
      </label>
      <div className={"control is-child"}>
        <input
          id={name}
          type={type}
          className={`${name} input`}
          name={name}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
        />
      </div>
    </div>
  );
};

export default RegistrationInputField;
