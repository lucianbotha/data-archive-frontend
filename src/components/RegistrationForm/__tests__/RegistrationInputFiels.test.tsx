import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RegistrationInputField from "../RegistrationInputField";

describe("RegistrationInputField Component", () => {
  it("renders the RegistrationInputField having unpopulated props with no errors", () => {
    const wrapper = shallow(<RegistrationInputField />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the RegistrationInputField having populated props with no errors", () => {
    const wrapper = shallow(
      <RegistrationInputField
        name={"username"}
        label={"Username"}
        value={""}
        onChange={() => {}}
        placeholder={""}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
