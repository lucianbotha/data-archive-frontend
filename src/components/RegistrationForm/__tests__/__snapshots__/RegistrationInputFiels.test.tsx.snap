// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationInputField Component renders the RegistrationInputField having populated props with no errors 1`] = `
<div
  className="field"
>
  <label
    className="label"
    htmlFor="username+Username"
  >
    Username
    :
  </label>
  <div
    className="control is-child"
  >
    <input
      className="username input"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder=""
      value=""
    />
  </div>
</div>
`;

exports[`RegistrationInputField Component renders the RegistrationInputField having unpopulated props with no errors 1`] = `
<div
  className="field"
>
  <label
    className="label"
    htmlFor="undefined+undefined"
  >
    :
  </label>
  <div
    className="control is-child"
  >
    <input
      className="undefined input"
    />
  </div>
</div>
`;
